src/components/directory-content.jsx

```diff
@@ -304,83 +266,84 @@ const DirectoryContent = () => {
-        (event) => {
-            const element = currentChildren?.find(
-                (e) => e.elementUuid === event.rowData?.elementUuid
-            );
-
-            if (selectedDirectory) {
-                dispatch(setActiveDirectory(selectedDirectory.elementUuid));
-            }
+    const getRowHovered = () => {
+        return gridRef.current?.api
+            ?.getRenderedNodes()
+            .find((node) => node.hovered);
+    };
 
-            if (element && element.uploading !== null) {
-                if (element.type !== 'DIRECTORY') {
+    const onCellContextMenu = useCallback(
+        (event, row) => {
+            if (row.data && row.data.uploading !== null) {
+                if (row.data.type !== 'DIRECTORY') {
                     setActiveElement({
                         hasMetadata:
-                            childrenMetadata[event.rowData.elementUuid] !==
+                            childrenMetadata[row.data.elementUuid] !==
                             undefined,
                         specificMetadata:
-                            childrenMetadata[event.rowData.elementUuid]
+                            childrenMetadata[row.data.elementUuid]
                                 ?.specificMetadata,
-                        ...element,
+                        ...row.data,
                     });
-
                     if (contextualMixPolicy === contextualMixPolicies.BIG) {
                         // If some elements were already selected and the active element is not in them, we deselect the already selected elements.
-                        if (
-                            selectedUuids?.size &&
-                            element?.elementUuid &&
-                            !selectedUuids.has(element.elementUuid)
-                        ) {
-                            setSelectedUuids(new Set());
+                        if (isRowUnchecked(row, checkedRows)) {
+                            setCheckedRows([]);
+                            gridRef.current?.api.deselectAll();
                         }
                     } else {
                         // If some elements were already selected, we add the active element to the selected list if not already in it.
-                        if (
-                            selectedUuids?.size &&
-                            element?.elementUuid &&
-                            !selectedUuids.has(element.elementUuid)
-                        ) {
-                            let updatedSelectedUuids = new Set(selectedUuids);
-                            updatedSelectedUuids.add(element.elementUuid);
-                            setSelectedUuids(updatedSelectedUuids);
+                        if (isRowUnchecked(row, checkedRows)) {
+                            setCheckedRows((prevCheckedRows) => [
+                                ...prevCheckedRows,
+                                formatMetadata(row.data, childrenMetadata),
+                            ]);
```

Comment:
is it necessary ? I think "handleRowSelected" is called when row selection is changing and should be done already

Reply:
Not needed indeed, simplifies a lot